crates/gpui/src/window.rs

```diff
@@ -3060,11 +3062,13 @@ impl<'a> WindowContext<'a> {
             }
             PlatformInput::MouseDown(mouse_down) => {
                 self.window.mouse_position = mouse_down.position;
+                self.window.mouse_down = true;
```

Comment:
I don't think this solution is sufficient yet, as the mouse exiting the window, releasing, and then re-entering might cause this `mouse_down` state to desync from the real mouse state. Maybe set it on mouse move too? We might need other events, let me know what you find :)